sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RetriableWorkItem.java

```diff
@@ -13,39 +15,50 @@
  */
 class RetriableWorkItem {
     private final AtomicInteger retryAttempts = new AtomicInteger();
-    private final MonoSink<Void> monoSink;
+    private final MonoSink<DeliveryState> monoSink;
     private final TimeoutTracker timeoutTracker;
     private final byte[] amqpMessage;
     private final int messageFormat;
     private final int encodedMessageSize;
+    private final AmqpTransaction transactionId;
 
     private boolean waitingForAck;
     private Exception lastKnownException;
 
-    RetriableWorkItem(byte[] amqpMessage, int encodedMessageSize, int messageFormat, MonoSink<Void> monoSink,
-                      Duration timeout) {
-        this(amqpMessage, encodedMessageSize, messageFormat, monoSink, new TimeoutTracker(timeout, false));
+    RetriableWorkItem(byte[] amqpMessage, int encodedMessageSize, int messageFormat, MonoSink<DeliveryState> monoSink,
+                      Duration timeout, AmqpTransaction transactionId) {
+        this(amqpMessage, encodedMessageSize, messageFormat, monoSink, new TimeoutTracker(timeout,
+            false), transactionId);
     }
 
-    private RetriableWorkItem(byte[] amqpMessage, int encodedMessageSize, int messageFormat, MonoSink<Void> monoSink,
-                              TimeoutTracker timeout) {
+    private RetriableWorkItem(byte[] amqpMessage, int encodedMessageSize, int messageFormat, MonoSink<DeliveryState>
+        monoSink, TimeoutTracker timeout, AmqpTransaction transactionId) {
```

Comment:
`AmqpTransaction` should be `transaction`, not `transactionId`. Same with other usages of this.